### PR TITLE
MAINT: updated github PR templates

### PR DESCRIPTION
**Pull request by samir-nasibli**
Changes:
- new template
- PR checklist job which fails if PR is not draft and hasn't all closed check boxes

# New template

## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
\ No newline at end of file
+Fixes # - _issue number(s) if exists_
+
+- [ ] I have reviewed my changes thoroughly before submitting this pull request.
+- [ ] I have commented my code, particularly in hard-to-understand areas.
+- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
+- [ ] The unit tests pass successfully.
+- [ ] I have run it locally and tested the changes extensively.
+- [ ] I have resolved any merge conflicts that might occur with the base branch.
+- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneDAL/blob/main/CONTRIBUTING.md#pull-requests) for details)_
```
Comment (napetrov) [inline .github/pull_request_template.md:12]: Let's use relative link instead for absolute.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
-Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.
+### Description 
+_Add a comprehensive description of proposed changes_
 
-Changes proposed in this pull request:
--
--
--
\ No newline at end of file
+Fixes # - _issue number(s) if exists_
```
Comment (napetrov) [inline .github/pull_request_template.md:4]: i would suggest you to split this to several different templates - MKL as an example - https://github.com/oneapi-src/oneMKL/issues/new/choose

Comment (samir-nasibli) [reply .github/pull_request_template.md:4]: Addressed on #2903

**Comment by samir-nasibli** [comment]
hm, I didn't expect that doc update will trigger jobs for library build and tests. Should be fixed

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**